### PR TITLE
[BREAKING] Replace `--split` with `--merge-sources` and Move generated extensions into their own source files

### PR DESCRIPTION
**Pull request by LePips**
- Overtakes https://github.com/CreateAPI/CreateAPI/pull/87
- Closes https://github.com/CreateAPI/CreateAPI/issues/74
- Part of https://github.com/CreateAPI/CreateAPI/issues/60

## What
- Makes splitting files the default and makes a new `--merge-sources` flag
- Creates separate extension files per purpose at the top level instead of merging them
- Updates tests to reflect

### CODE REVIEW

**Review by liamnichols — COMMENTED** [review]
Great work on this, thanks for picking it up @LePips! 

What is the notification however for moving the extensions out of **Entities.swift** and **Paths.swift** when **using `--merge-sources`**? It makes sense when splitting, but this one looks a little odd:

Screenshot 2022-08-04 at 20 42 28

I think I would have expected these to stay inside the merged file?

---

Secondly, I just want to bring this to your attention: https://github.com/CreateAPI/CreateAPI/issues/47

I'm thinking that maybe we can just go all of the way and move `--merge-sources` into the configuration options instead? WDYT about that approach? 

There is also another part of that ticket that is relevant to you, which is about how to go about describing the filename templates in the config too so any feedback is welcomed 👍

**Comment by LePips** [comment]
> extension using `--merge-sources`

I was under the impression that they should be broken out no matter what from https://github.com/CreateAPI/CreateAPI/issues/60 and personally think they should be. The only use case I can think of for merging schema-generated objects/paths would be for reference. While devs might also move files around, they will still have to be responsible for moving these extensions as well (split or merged).

Essentially, it splits the concerns of each file: `Entities.swift` for generated objects, `Paths.swift` for generated paths, and then each extension file is their own helper file.

> #47

I don't think that this simple change in logic should worry about that until a consensus is reached there. I'll leave my thoughts on the issue.

**Comment by liamnichols** [comment]
> I was under the impression that they should be broken out no matter wha

Doh, you are right.. I forgot that I even suggested that 😅  https://github.com/CreateAPI/CreateAPI/issues/60#issuecomment-1203141626 

> I don't think that this simple change in logic should worry about that until a consensus is reached there. I'll leave my thoughts on the issue.

Sure 👍 

---

Let me just do some final tests of this change 👀 It's hard to review on GitHub as the diff freezes up my browser 😄

File: Sources/CreateAPI/Generate.swift [hunk 1]
```diff
@@ -20,8 +20,11 @@ struct Generate: ParsableCommand {
     @Option(help: "The path to generator configuration. If not present, the command will look for .create-api.yaml in the current directory.")
     var config = "./.create-api.yaml"
 
-    @Flag(name: .shortAndLong, help: "Split output into separate files")
-    var split = false
+    @Flag(help: ArgumentHelp("Merge Entities and Paths into single output files", discussion: """
+                            Merging the source files offers a compact output, but prevents the compiler \
+                            from parallelizing build tasks resulting in slower builds for larger schemas.
+                            """))
```
Comment (liamnichols) [inline Sources/CreateAPI/Generate.swift:26]: Should only be one level of indentation here

```suggestion
    @Flag(help: ArgumentHelp("Merge Entities and Paths into single output files", discussion: """
        Merging the source files offers a compact output, but prevents the compiler \
        from parallelizing build tasks resulting in slower builds for larger schemas.
        """))
```

**Review by liamnichols — COMMENTED** [review]
It looks like the old **Tests/CreateAPITests/Expected/petstore-split** directory is still here, can we delete it please 🙏

**Comment by liamnichols** [comment]
~Please can you revert the rest of the changes to **Snapshots.swift** 🙏~ Keep in mind that this diff is gigantic and barely loads in the browser so it makes it hard to efficiently review things 🙇

Its not worth changing it now, but my suggestion if we were to do this again would be to submit a pull request that switches `--split` for `--merge-sources`, but updates all the test fixtures that didn't use `--split` to use `--merge-sources` (and vice versa).. This would have helped us to be completely sure that the only other changes in output were related to moving the extensions since the diff would have been readable. We could then update all the fixtures to stop using `--merge-sources` in a separate PR where the diff is less important 🙏 

Thanks again though for working on this! Will merge it once the checks finish ✅

**Comment by liamnichols** [comment]
Thanks again @LePips for the great contribution 🚀

**Pull request merged**